pyfolio/tears.py

```diff
@@ -1217,3 +1229,161 @@ def create_bayesian_tear_sheet(returns, benchmark_rets=None,
+                           volumes=None,
+                           percentile=None,
+                           return_fig=False):
+    '''
+    Creates risk tear sheet: computes and plots style factor exposures, sector
+    exposures, market cap exposures and volume exposures.
+
+    Parameters
+    ----------
+    positions : pd.DataFrame
+        Daily equity positions of algorithm, in dollars.
+        - DataFrame with dates as index, equities as columns
+        - Last column is cash held
+        - Example:
+                     Equity(24   Equity(62
+                       [AAPL])      [ABT])             cash
+        2017-04-03	-108062.40 	  4401.540     2.247757e+07
+        2017-04-04	-108852.00	  4373.820     2.540999e+07
+        2017-04-05	-119968.66	  4336.200     2.839812e+07
+
+    style_factor_panel : pd.Panel
+        Panel where each item is a DataFrame that tabulates style factor per
+        equity per day.
+        - Each item has dates as index, equities as columns
+        - Example item:
+                     Equity(24   Equity(62
+                       [AAPL])      [ABT])
+        2017-04-03	  -0.51284     1.39173
+        2017-04-04	  -0.73381     0.98149
+        2017-04-05	  -0.90132	   1.13981
+
+    sector : pd.DataFrame
+        Daily Morningstar sector code per asset
+        - DataFrame with dates as index and equities as columns
+        - Example:
+                     Equity(24   Equity(62
+                       [AAPL])      [ABT])
+        2017-04-03	     311.0       206.0
+        2017-04-04	     311.0       206.0
+        2017-04-05	     311.0	     206.0
+
+    caps : pd.DataFrame
+        Daily Morningstar sector code per asset
+        - DataFrame with dates as index and equities as columns
+        - Example:
+                          Equity(24        Equity(62
+                            [AAPL])           [ABT])
+        2017-04-03     1.327160e+10     6.402460e+10
+        2017-04-04	   1.329620e+10     6.403694e+10
+        2017-04-05	   1.297464e+10	    6.397187e+10
+
+    shares_held : pd.DataFrame
+        Daily number of shares held by an algorithm.
+        - Example:
+                          Equity(24        Equity(62
+                            [AAPL])           [ABT])
+        2017-04-03             1915            -2595
+        2017-04-04	           1968            -3272
+        2017-04-05	           2104            -3917
+
+    volumes : pd.DataFrame
+        Daily volume per asset
+        - DataFrame with dates as index and equities as columns
+        - Example:
+                          Equity(24        Equity(62
+                            [AAPL])           [ABT])
+        2017-04-03      34940859.00       4665573.80
+        2017-04-04	    35603329.10       4818463.90
+        2017-04-05	    41846731.75	      4129153.10
+
+    percentile : float
+        Percentile to use when computing and plotting volume exposures.
+        - Defaults to 10th percentile
+    '''
+
```

Comment:
You should also pass `positions` through `check_intraday`, like here: https://github.com/quantopian/pyfolio/blob/master/pyfolio/tears.py#L603

Otherwise, for strategies that have no end-of-day holdings, your tear sheet will probably not be able to plot much.

That also requires passing transactions and returns to `create_risk_tear_sheet` :)